usr/src/lib/libzfs/common/libzfs_dataset.c

```diff
@@ -3660,7 +3660,7 @@ zfs_promote(zfs_handle_t *zhp)
 			return (zfs_error(hdl, EZFS_EXISTS, errbuf));
 
 		default:
-			return (zfs_standard_error(hdl, save_errno, errbuf));
+			return (zfs_standard_error(hdl, ret, errbuf));
```

Comment:
Yep.  And since we are removing save_errno, I don't think it will compile without this additional change.